ISL.Providers.Storages.AzureBlobStorage.Tests.Unit/Services/Foundations/Files/StorageTests.Validations.RetrieveFile.cs

```diff
@@ -0,0 +1,60 @@
+﻿using FluentAssertions;
+using ISL.Providers.Storages.AzureBlobStorage.Models.Foundations.Files.Exceptions;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace ISL.Providers.Storages.AzureBlobStorage.Tests.Unit.Services.Foundations.Files
+{
+    public partial class StorageTests
+    {
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(null, "")]
+        [InlineData(null, " ")]
+        [MemberData(nameof(GetStreamWithLength))]
+        public async Task ShouldThrowValidationExceptionOnRetrieveIfArgumentsInvalidAsync(Stream invalidStream, string invalidText)
+        {
+            // given
+            string invalidFileName = invalidText;
+            string invalidContainer = invalidText;
+            Stream invalidInputStream = invalidStream;
```

Comment:
Use this instead as you need to validate both conditions.
1) Cant be null
2) You don't want to stream to a stream that aleady has data on it.  We are not appending.

```suggestion
        [Theory]
        [InlineData(null, null)]
        [InlineData("empty", "")]
        [InlineData("empty", " ")]
        [MemberData(nameof(GetStreamWithLength))]
        public async Task ShouldThrowValidationExceptionOnRetrieveIfArgumentsInvalidAsync(Stream invalidStream, string invalidText)
        {
            // given
            string invalidFileName = invalidText;
            string invalidContainer = invalidText;
            Stream invalidInputStream = invalidStreamType == null ? null : new MemoryStream();
```